src/Components/Toast.php

```diff
@@ -25,9 +26,22 @@ public function handleIncomingMessage(array $message): void
         $this->toasts[uniqid()] = [
             'message' => $message[0],
             'type'    => $message[1],
+            'style'    => $this->getValidStyle(Arr::get($message, 2, 'regular')),
         ];
     }
 
+    private function getValidStyle(string $style): string
+    {
+        return Arr::get([
+            // Without close button and icon
+            'simple' => 'simple',
+            // Only icon, no close button
+            'onlyicon' => 'onlyicon',
+            // With close button and icon
+            'regular' => 'regular',
```

Comment:
this would be simpler if you could pass `withIcon` and `withButton` so you can mix and match what you need 🤔 no need to define styles in that case as you can customize it by setting `withIcon=false` and having everything enabled by default.

Reply:
Besides, toasts should always be the same. There is an alert component that has different styles but a toast should always have an icon + text + close button or whatever is shown in the design. There shouldn't be different toast designs used afaik

Reply:
Is there a reason you're not passing the configuration as an array / object to the event instead of an array with "hardcoded" indexes (first index message, second style, etc)? Just more robust than having to know which index represents which parameter.

```javascript
Livewire.emit('toastMessage', {
    message: 'Hello world',
    state: 'info',
    // icon: false,
    // close: false
})
```

```php
$this->emit('toastMessage', [
    'message' => 'Hello world',
    'state' => 'info',
    // 'icon' => false,
    // 'close' => false,
]);

// Could be parsed that if string, default to a certain state of toast.

$this->emit('toastMessage', 'Hello world');
```

Reply:
@ItsANameToo removed custom styles and updated ticket description accordingly